### PR TITLE
WIP : Changes for remote translog backed by blob store

### PR DESCRIPTION
**Pull request by Bukhtawar**
Signed-off-by: Bukhtawar Khan <bukhtawa@amazon.com>

### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/index/IndexService.java [hunk 1]
```diff
@@ -553,6 +559,10 @@ public synchronized IndexShard createShard(
                 () -> globalCheckpointSyncer.accept(shardId),
                 retentionLeaseSyncer,
                 circuitBreakerService,
+                this.indexSettings.isSegRepEnabled() && this.indexSettings.isRemoteStoreEnabled()
+                    ? new RemoteBlobStoreInternalTranslogFactory(
+                        repositoriesServiceSupplier.get().repository(clusterService.state().metadata().clusterUUID()))
+                    : new InternalTranslogFactory(),
```
Comment (getsaurabh02) [inline server/src/main/java/org/opensearch/index/IndexService.java:565]: Can we skip the resolution of ``repositoriesServiceSupplier`` here unless needed by the IndexService, and instead let the ``RemoteBlobStore`` resolve it when necessary?

File: server/src/main/java/org/opensearch/index/translog/LocalTranslog.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            }
+            boolean success = false;
+            current = null;
+            try {
+                current = createWriter(
+                    checkpoint.generation + 1,
+                    getMinFileGeneration(),
+                    checkpoint.globalCheckpoint,
+                    persistedSequenceNumberConsumer
+                );
```
Comment (getsaurabh02) [inline server/src/main/java/org/opensearch/index/translog/LocalTranslog.java:88]: Is the previously deleted file ``nextTranslogFile`` always unusable, post every ``recoverFromFiles`` ?

**Comment by dreamer-89** [comment]
@Bukhtawar : Are you still working on this ?

**Comment by andrross** [comment]
Closing as remote translog has been implemented. @Bukhtawar please reopen if you still need any of this.

**Pull request closed without merging**